controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +590,109 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			if err != nil {
 				return ctrl.Result{}, err
 			}
+			// Delete DNS thingy
+			dnsRecord := &infranetworkv1.DNSData{}
+			dnsName := "dns-" + serviceName + fmt.Sprintf("-%d", i)
+			dnsNamespace := helper.GetBeforeObject().GetNamespace()
+			Log.Info(fmt.Sprintf("Name: %v, Namespace: %v", dnsName, dnsNamespace))
+			test_err := helper.GetClient().Get(ctx, types.NamespacedName{Name: dnsName, Namespace: dnsNamespace}, dnsRecord)
+			if test_err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", test_err))
+			}
+			err := helper.GetClient().Delete(ctx, dnsRecord)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", err))
+			}
+
 		}
 	}
+
+	Log.Info("DNS Starts here")
+
+	serviceName = ovndbcluster.ServiceNameSB
+	if instance.Spec.DBType == v1beta1.NBDBType {
+		serviceName = ovndbcluster.ServiceNameNB
+	}
+	serviceLabels = map[string]string{
+		common.AppSelector: serviceName,
+	}
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if len(podList.Items) > int(*(instance.Spec.Replicas)) {
+
+	}
+	for i, ovnPod := range podList.Items {
+		if i >= int(*(instance.Spec.Replicas)) {
+			Log.Info(fmt.Sprintf("In iteration %v is OOO, exit", i))
+			break
+		}
+		Log.Info(fmt.Sprintf("Using Pod: %v", ovnPod.Name))
+		var dnsName string
+		var dnsIP string
+		var hostnames []string
+		dnsName = "dns-" + ovnPod.Name
```

Comment:
may be better to create 1 dnsdata per DBCluster instead of per pod?

Reply:
It may complicate logic when scaling up/down. I can't see any benefit of having only 1 DNSData per cluster instead per service.

Reply:
With one per pod will scale down need explicit logic to delete the DNSData?

Reply:
Well, would need new logic to retrieve the DNSData, iterate over all hosts and delete those which had been scaled-down and update the resource. 
Currently I just delete the DNSData CR associated to the scaled-down pod/service

Reply:
For this case i.e with keeping 1 dnsdata per DBCluster, i think not required to fetch/delete but can just update dnsdata with current set of pods in case of scale up/down.